tests/perf/benchmark.py

```diff
@@ -151,138 +161,146 @@ def run(
-            self._log_metrics(
-                work_dir=sub_work_dir / ".latest" / "train",
-                tags=tags,
-                criteria=criteria,
-                extra_metrics=extra_metrics,
-            )
-
-            command = [
-                "otx",
-                "test",
-                "--work_dir",
-                str(sub_work_dir),
-            ]
-            for key, value in dataset.extra_overrides.get("test", {}).items():
-                command.append(f"--{key}")
-                command.append(str(value))
-            self._run_command(command)
-            self._rename_raw_data(
-                work_dir=sub_work_dir / ".latest" / "test",
-                replaces={"test_": "test/", "{pre}": "test/"},
-            )
-            self._log_metrics(work_dir=sub_work_dir / ".latest" / "test", tags=tags, criteria=criteria)
-
-            # Export & test
-            if self.eval_upto in ["export", "optimize"]:
-                command = [
-                    "otx",
-                    "export",
-                    "--work_dir",
-                    str(sub_work_dir),
-                ]
-                for key, value in dataset.extra_overrides.get("export", {}).items():
-                    command.append(f"--{key}")
-                    command.append(str(value))
-                self._run_command(command)
-
-                exported_model_path = sub_work_dir / ".latest" / "export" / "exported_model.xml"
-                if not exported_model_path.exists():
-                    exported_model_path = sub_work_dir / ".latest" / "export" / "exported_model_decoder.xml"
+            try:
+                sub_work_dir = work_dir / str(seed)
+                tags["seed"] = str(seed)
 
-                command = [  # NOTE: not working for h_label_cls. to be fixed
+                # Train & test
+                command = [
                     "otx",
-                    "test",
-                    "--checkpoint",
-                    str(exported_model_path),
+                    "train",
+                    "--config",
+                    f"src/otx/recipe/{model.task}/{model.name}.yaml",
+                    "--data_root",
+                    str(data_root),
                     "--work_dir",
                     str(sub_work_dir),
+                    "--engine.device",
+                    self.accelerator,
                 ]
-                for key, value in dataset.extra_overrides.get("test", {}).items():
+                for key, value in dataset.extra_overrides.get("train", {}).items():
                     command.append(f"--{key}")
                     command.append(str(value))
+                command.extend(["--seed", str(seed)])
+                # TODO(someone): Disable deterministic for instance segmentation as it causes OOM.
+                # https://github.com/pytorch/vision/issues/8168#issuecomment-1890599205
+                command.extend(["--deterministic", str(self.deterministic)])
```

Comment:
@chuneuny-emily There's an issue in RoIAlign where we could get OOM training on CUDA when deterministic is enabled. I'd suggest turning off deterministic for inst-seg if you experience any CUDA OOM in the future until deterministic is fixed in RoIAlign.

Reply:
@eugene123tw - The current default setting appears to be deterministic=False. But when I check the benchmark code, there is an issue in the extra_overrides logic for both instance segmentation and detection, where deterministic is redundantly set to True. This PR aims to address another issue where tests are not completing across all seeds due to exceptions, by adding try-catch blocks and excluding actions from the test scope. Since there have been no CUDA OOM issue with the existing ins_seg performance measurements before, and the support for deterministic is not clear across different algorithms, it is suggested to defer changes until the next PR.